sepal_ui/mapping/legend.py

```diff
@@ -0,0 +1,112 @@
+from ipywidgets import HTML
+from traitlets import Bool, Dict, Unicode, observe
+
+import sepal_ui.sepalwidgets as sw
+from sepal_ui.message import ms
+from sepal_ui.scripts import utils as su
+
+
+class Legend(sw.Card):
+    """
+    Custom Card to display and update map legend
+
+    """
+
+    title = Unicode("Legend").tag(sync=True)
+    "str: title of the legend. defaults to 'Legend'"
+
+    legend_dict = Dict({}).tag(sync=True)
+    "dict: dictionary with key as label name and value as color"
+
+    vertical = Bool(True).tag(sync=True)
+    "bool: whether to display the legend in a vertical or horizontal way"
+
+    _html_table = None
+
+    def __init__(self, legend_dict, title=ms.mapping.legend, vertical=True, **kwargs):
+
+        # Be sure that the scroll bar will be shown up when legend horizontal
+        self.style_ = "overflow-x:auto; white-space: nowrap;"
+        self.max_width = 450
+        self.max_height = 350
+        self.title = title
+        self.legend_dict = legend_dict
+        self.vertical = vertical
+
+        self.html_title = sw.Html(tag="h4", children=[f"{self.title}"])
+        self._html_table = sw.Html(tag="table", children=[])
+        self.children = [self.html_title, self._html_table]
+
+        super().__init__(**kwargs)
+
+        self.set_legend(None)
+
+    def __len__(
+        self,
+    ):
+        """returns the number of elements in the legend"""
+        return len(self.legend_dict)
+
+    @observe("legend_dict", "vertical")
+    def set_legend(self, _):
```

Comment:
don't you think it could make sense to raise an error if the legend_dict is empty ?